### ISSUE TITLE
[Bug] Torch `random.shuffle(axis=-1)` mishandles negative axes

### ISSUE THREAD

**Issue by ChiragSW** [keras-team/keras#23118]
### Description

`keras.random.shuffle(..., axis=-1)` produces incorrect results with the Torch backend.

Negative axes are not canonicalized before building shuffle indices, so `axis=-1` is treated as the literal axis value instead of the last dimension. 

### Minimal reproduction

```python
import os
os.environ["KERAS_BACKEND"] = "torch"

import keras
import numpy as np

x = keras.ops.reshape(keras.ops.arange(12), (3, 4))
y = keras.random.shuffle(x, axis=-1, seed=123)

x_np = keras.ops.convert_to_numpy(x)
y_np = keras.ops.convert_to_numpy(y)

print(x_np)
print(y_np)

for before, after in zip(x_np, y_np):
    assert sorted(after.tolist()) == sorted(before.tolist()), (before, after)
```
### Expected behavior

axis=-1 should behave the same as axis=x.ndim - 1

There is need of canonicalization in `src/backend/torch/random.py` in the shuffle implementation.
